File: 4.Counting Element - PermCheck/answer.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    intDict = {i: None for i in range(1, len(A)+1)}
    
    for i in A:
        if(i in intDict):
            del intDict[i]
        else:
            return 0
    if(len(intDict) > 0):
        return 0
        
    return 1

File: 15.Caterpillar Method - AbsDistinct/answer.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    let left = 0, right = A.length-1
    let numSet = new Set()
    while(left <= A.length/2){
        numSet.add(Math.abs(A[left]))
        numSet.add(Math.abs(A[right]))
        left++
        right--
    }
    
    return numSet.size
}
